src/services/newMessagesService.ts

```diff
@@ -310,6 +360,69 @@ export default class NewMessagesService {
       TE.toUnion
     )();
 
+  // ------------------------------------
+
+  private readonly getThirdPartyMessageFnApp = (
+    fiscalCode: FiscalCode,
+    messageId: string
+  ): TE.TaskEither<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorForbiddenNotAuthorized
+    | IResponseErrorNotFound
+    | IResponseErrorTooManyRequests,
+    MessageWithThirdPartyData
+  > =>
+    pipe(
+      TE.tryCatch(
+        () =>
+          this.apiClient.getMessage({
+            fiscal_code: fiscalCode,
+            id: messageId
+          }),
+        e => ResponseErrorInternal(E.toError(e).message)
+      ),
+      TE.chain(wrapValidationWithInternalError),
+
+      TE.chainW(
+        flow(
+          response =>
+            response.status === 200
+              ? E.of(response.value.message)
+              : E.left(response),
+          TE.fromEither,
+          TE.mapLeft(response => {
+            log.error(
+              `newMessagesServixce|getThirdPartyMessageFnApp|result:${
+                response.status
+              }(${response.value ?? "No details"})`
+            );
+            return response;
+          }),
+          TE.mapLeft(response => {
+            switch (response.status) {
+              case 401:
+                return ResponseErrorUnexpectedAuthProblem();
+              case 404:
+                return ResponseErrorNotFound("Not found", "Message not found");
+              case 429:
+                return ResponseErrorTooManyRequests();
+              default:
+                return ResponseErrorStatusNotDefinedInSpec(response);
+            }
+          })
+        )
+      ),
+      TE.chainW(
+        // MessageWithThirdPartyData.is fails, we need to check the decode instead
+        TE.fromPredicate(isMessageWithThirdPartyData, () =>
+          ResponseErrorInternal(
```

Comment:
In this case we can return Bad Request instead

Reply:
Sure

Reply:
Done in [78754ae](https://github.com/pagopa/io-backend/pull/899/commits/78754aee52cd910daf09259253fee5521f3792bd)